### ISSUE TITLE
class:list will be sorted

### ISSUE THREAD

**Issue by J-env** [withastro/astro#8108]
### What version of `astro` are you using?

2.8.5

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

npm,pnpm

### What operating system are you using?

Mac

### What browser are you using?

Chrome

### Describe the Bug

If there are numbers in the php statement, the class name will be sorted.

<img width="792" alt="image" src="https://github.com/withastro/astro/assets/25259920/bec3194f-0aaa-4b12-ba88-f64857d3c6b7">

build:
<img width="756" alt="image" src="https://github.com/withastro/astro/assets/25259920/319d1643-be54-4463-a57a-b9407ee2eb99">

### What's the expected result?

can it be kept as it is

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-sdeict-afdy3k?file=src%2Fpages%2Findex.astro,dist%2Findex.html

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by natemoo-re** [comment]
I investigated this a bit. Looks like our implementation diverges from `clsx` because we try to deduplicate identical classes. Surprised to see that Astro sorts the keys considering that we don't call `sort` anywhere in the code.

https://stackblitz.com/edit/github-sdeict-hvpa6y?file=src%2Fpages%2Findex.astro,src%2Fcomponents%2FTest.astro

A year ago we discussed replacing our implementation with `clsx` directly but opted against it for backwards-compatability reasons. Maybe it's time to pull the trigger on that for 3.0.

https://github.com/withastro/astro/pull/3935
